File: Practice/atcoder/ABC/125/src/d.py
# できるだけ正数が多いほうが良い。隣り合う負数は正にできる
# あと-+++++は+++++-のように-を移動できるので隣り合わせられる
# -が奇数なら0以上の最小の数に-を割り当ててあとは合計
# 偶数ならすべての-を+にできる

from functools import reduce

n = int(input())
minn = 10**9
arr = list(map(int, input().split()))
minus = 0
ans = reduce(lambda x, y: abs(x) + abs(y), arr)
for i in range(n):
    if arr[i] < 0:
        minus += 1
    if minn > abs(arr[i]):
        minn = abs(arr[i])
if minus % 2 == 0:
    print(ans)
else:
    print(ans-minn*2)

File: Practice/atcoder/ABC/125/src/d.go
// 問題文読んでないけど負数が奇数の場合は両端どっちかで偶数の場合はなんとなく全部正にできそう
// """0""" の存在を忘れていた...
// 意味不明なことにWA
// アホだった なーにが「両端どっちか」じゃボケが 最小値にすることが可能
package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)
	A := make([]int, N)
	S := 0
	minus := 0
	zero := false
	minn := int(1e10)
	for i := 0; i < N; i++ {
		var V int
		fmt.Scan(&V)
		if V == 0 {
			zero = true
		} else {
			minn = min(minn, abs(V))
		}
		A[i] = abs(V)
		S += abs(V)
		if V < 0 {
			minus++
		}
	}
	if zero {
		fmt.Println(S)
	} else if minus%2 == 0 {
		fmt.Println(S)
	} else {
		fmt.Println(S - minn*2)
	}
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func abs(x int) int {
	if x < 0 {
		x *= -1
	}
	return x
}
